vm/compiler/benches/coinbase_puzzle.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(clippy::single_element_loop)]
+
+#[macro_use]
+extern crate criterion;
+
+use console::{account::*, network::Testnet3};
+use snarkvm_compiler::{CoinbasePuzzle, EpochChallenge, PuzzleConfig};
+
+use criterion::Criterion;
+use rand::{self, thread_rng, CryptoRng, RngCore};
+
+type CoinbasePuzzleInst = CoinbasePuzzle<Testnet3>;
+
+fn sample_inputs(
+    degree: u32,
+    rng: &mut (impl CryptoRng + RngCore),
```

Comment:
It's probably a good idea to replicate the production environment in benches, no?

Reply:
that depends; in terms of marker traits it doesn't make a difference, but it could be helpful if some API needs to be adhered to; as for the source of randomness, `thread_rng` might provide more representative results, but it will reduce the usefulness of the benchmark as a means to find potential performance improvements (which are unlikely to be introduced on the randomness side in production)